[docs] Add a warning about to clear the local storage when `defaultMode` changes (#35937)
mui/material-ui: docs/data/joy/customization/dark-mode/dark-mode.md

To set dark mode as the default for your app, add `defaultMode: 'dark'` to your `<CssVarsProvider>` wrapper component:

:::warning

> Note that when you change the defaultMode to another value, you have to always clear the local storage.

Reviewer: ```suggestion
When you change the `defaultMode` to another value, you have to always clear the local storage to take effect.
```

> When you change the `defaultMode` to another value, you have to always clear the local storage to take effect.

Reviewer 2: Tiny grammar change for consistency.

```suggestion
When you change the `defaultMode` to another value, you must clear the local storage for it to take effect.
```
